### DISCUSSION TITLE
Wrong Weapon on Purchase Event

### DISCUSSION THREAD

**Discussion by ricardo177m** [saul/demofile#364] [Questions] [answered]
Hello. After using demofile I noticed that `IItemPurchaseEvent` does not return the correct weapon property for, as far as I know, Kevlar (weaponId 50) and Helmet (weaponId 51).

Thanks.

**Comment by saul** [comment]
What weapon is returned? Some items don't have an associated entity that sticks around after the player picks it up/purchases it.

**Reply by ricardo177m** [reply]
When a player buys a helmet, the weaponId returned is 50, the weapon object returns undefined and the itemName returned is "C4".

**Reply by saul** [reply]
PR with fix here: https://github.com/saul/demofile/pull/365

**Comment by saul** [comment, marked as answer]
Good catch, this is now fixed in 2.9.2

**Reply by ricardo177m** [reply]
Thank you! Glad to help.
